.github/workflows/main.yaml

```diff
@@ -0,0 +1,37 @@
+name: Pre-commit Checks
+
+on:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Source Code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
+
+      - name: Check License
+        run: make license-check
+
+      - name: Set Node.js Environment
+        uses: actions/setup-node@v2
+        with:
+          node-version: 12.14
+      - run: npm install -g @angular/cli  > /dev/null
+      - run: npm install karma
+      - run: npm install -g protractor
+      - run: yarn install
```

Comment:
I think this can be simplified to:

```
run: |
      npm install -g @angular/cli  > /dev/null
      npm install karma
      ...
```
the above syntax is supported in github action for multi-line commands.